### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by anandarabi** [kenoma/adaptive-neuro-fuzzy#9]
Please explain how are you calculating stand. deviation from the euclidean distance from the nearest neighbor in Gaussian Rule (and in Gaussian Rule 2) function
Why are you keeping same repeated (identical) values of stand. deviation (denoted by "a") in the end portion of your parameters array in Gaussian Rule. In Gaussian Rule 2, you are keeping only one stand. deviation value (denoted by "a")

**Comment by kenoma** [comment]
Hi!

When you initialize whole FIS you need to set some values for each membership function to start learning. My choice for doing was to cover whole n-dimensional manifold with overlaping membership functions and tune FIS parameters with backprop procedure.

Look at method `GaussianRule.Init` here you're setting parameter `a` as an distance to `NearestNeighb`.   
You can find that `NearestNeighb` is nearest centroid from an initial clusterization set (see `RuleSetFactory.Build`). After this you leave parameter `a` to the mercy of gradient descent.

>Why are you keeping same repeated (identical) values of stand. deviation (denoted by "a") in the end portion of your parameters array in Gaussian Rule. In Gaussian Rule 2, you are keeping only one stand. deviation value (denoted by "a")

It's an optimizational trick to easy gradient computations.

**Comment by anandarabi** [comment]
Thank you for the reply.

Please clarify on the computation of the value of "a" (standard deviation).
From nearest neighbour distance what you are up to please clarify...
You have used formula like
/*****/
            double desiredVatneigb =  1e-10;
            double d2 = math.EuclidianDistance2(Centroid, NearestNeighb);
            double a = Math.Sqrt(-d2 / (2 * Math.Log(desiredVatneigb)));

/*****/
Thank you so much once again sir!

**Comment by kenoma** [comment]
Ah, I see. When I'm covering manifold with initial set of membership functions I wish them to overlap each other in order to avoid situation when single internal point of manifold belongs only to one membership function. This case leads to vanishing gradient at this point and whole FIS can stuck in local optima at start of learning procedure. So, when I'm computing initial `a` values I suppose that at the _core_ of an membership function it will be non-zero firing level (with value`desiredVatneigb`) from nearest neighbourhood. I have found this technique quite universal for wide range of task, so I have implemented it.

**Comment by anandarabi** [comment]
Sir,
Thank you so much.
I wish if I could be in touch with highly professionals like you in
emails...
I express my sincere gratitude to you.
Regards!

**Comment by kenoma** [comment]
You are welcome. Feel free to raise discussions here, someone else can find it useful too.

**Comment by anandarabi** [comment]
Please note my email as anandarabi@gmail.com
and please drop your email so that we can communicate.

Thank you. Regards.
